pantheon-advanced-page-cache.php

```diff
@@ -18,6 +18,8 @@
  * @param array $keys Surrogate keys to purge.
  */
 function pantheon_wp_clear_edge_keys( $keys ) {
+  $keys = array_unique( $keys );
+  $keys = \Pantheon_Advanced_Page_Cache\Emitter::filter_huge_surrogate_keys_list( $keys );
```

Comment:
Why are you calling `filter_huge_surrogate_keys_list()` in this context?

Reply:
I'm not sure, if Pantheon purges the keys with Header or JSON post body. 

But after double checking the [Fastly Documentation](https://docs.fastly.com/api/purge#purge_db35b293f8a724717fcf25628d713583), it would be better to check for max 256 keys instead of `const HEADER_MAX_LENGTH = 32512` assuming they purge via JSON post body.

Reply:
> Do you want to finish up this PR or shall I?

Nevermind, just saw your other comments. We can land this today, and then I'll tag a release on Monday.

Reply:
> Pantheon has middleware to safely handle large lists of keys.
👍 
> Do you want to finish up this PR or shall I?
 Happy to hand it over. Otherwise I'll have a look at this weekend.

Reply:
Ok, doing now.